packages/expo-router/src/layouts/StackClient.tsx

```diff
@@ -27,8 +29,45 @@ type GetId = NonNullable<RouterConfigOptions['routeGetIdList'][string]>;
 
 const NativeStackNavigator = createNativeStackNavigator().Navigator;
 
+/**
+ * We extend NativeStackNavigationOptions with our custom props
+ * to allow for several extra props to be used on web, like modalWidth
+ */
+export type ExtendedStackNavigationOptions = NativeStackNavigationOptions & {
+  /**
+   * Override the width of the modal (px or percentage). Only applies on web platform.
+   * @platform web
+   */
+  modalWidth?: number | string;
```

Comment:
Let's check if there are already options that set the modal width on iOS/Android and try to reuse them before adding new single-platform variables.